src/Tutor.Web/Startup.cs

```diff
@@ -235,22 +237,22 @@ private void SetupServices(IServiceCollection services)
         SetupLearningServices(services);
         SetupManagementServices(services);
 
-        services.AddScoped<IUnitAnalysisService, UnitAnalysisService>();
+        services.AddProxiedScoped<IUnitAnalysisService, UnitAnalysisService>();
     }
 
     private void SetupLearningServices(IServiceCollection services)
     {
-        services.AddScoped<ISessionService, SessionService>();
-        services.AddScoped<IStructureService, StructureService>();
-        services.AddScoped<IStatisticsService, StatisticsService>();
-        services.AddScoped<ISelectionService, SelectionService>();
-        services.AddScoped<IEvaluationService, EvaluationService>();
-        services.AddScoped<IHelpService, HelpService>();
+        services.AddProxiedScoped<ISessionService, SessionService>();
```

Comment:
Can you briefly explain this change?

Reply:
Instead of registering services directly to do the ASP.NET service container, we add a Proxy level in between in order to intercept all calls to service methods. Multiple proxies can be added, currently, we have only the Logger.
Sample of execution is given in the picture.

![Alt text](https://raw.githubusercontent.com/castleproject/Core/master/docs/images/proxy-pipeline.png "Execution example")